ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -120,8 +123,13 @@ public void start() throws StoreException {
 
         StoreDescriptor storeDescriptor = new StoreDescriptor(dataDir);
         log = new Log(dataDir, capacityInBytes, config.storeSegmentSizeInBytes, metrics);
+        compactor =
+            new BlobStoreCompactor(dataDir, storeId, factory, config, metrics, diskIOScheduler, log, taskScheduler,
+                recovery, time, sessionId, storeDescriptor.getIncarnationId());
+        compactor.fixStateIfRequired();
```

Comment:
Yeah this is a good suggestion. Will move it to the constructor. All the info needed for fixing will have been initialized.